Sources/JoyfillComponents/Components/Chart/ChartView.swift

```diff
@@ -474,26 +495,26 @@ public class ChartView: UIViewController, UITextFieldDelegate, ChartViewTextFiel
     
     // ScrollView delegate method calls on scroll
     public func scrollViewDidScroll(_ scrollView: UIScrollView) {
-        reloadTimer?.invalidate()
-        reloadTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) { [weak self] _ in
-            self?.addLineTableView.reloadData()
+        if !isTextFieldSelected {
+            reloadTimer?.invalidate()
+            reloadTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) { [weak self] _ in
+                self?.addLineTableView.reloadData()
```

Comment:
So for this, what does this do and does it fix to have a 1 second timer, I know that code like this is very discouraged to use timers especially for small UI fixes. I am not apposed to it I would just like to know why and was there no other way to resolve this in a cleaner way?

Reply:
Actually this is related to reload tableView and update the values inside it. This is not related to UI fixes.
If we call it without timer then it will show lag while scrolling. So, to fix that lag this timer is used here.

Reply:
Okay but why would it cause lag? And why does the 1 second delay fix the lag? Also does Swift auto destroy the timer or do all those created timers just stay there?

Reply:
Because it will repeatedly refresh tableView if we call it without any time, this will cause slowness. This means that all of the tableView delegate methods will also be called, which will lag and consume time.

Yes, this is the reason why the tableField condition is used; upon the user's click in any input field, the timer will end.